### DISCUSSION TITLE
Question: How to format data from a measurement before returning it?

### DISCUSSION THREAD

**Discussion by jwbrooks0** [pymeasure/pymeasure#898] [Q&A] [answered]
For my device, I've created several Instrument.measurement() that returns data from my device.  In order to be immediately useful, the data requires some formatting or processing before returning it.  

In my first example, the data comes back as a list of floats and I want it instead to be a numpy array.  If I were naively processing this data, I would write a function like this:
``` 
import numpy as np
def _process(self, x):
    return np.array(x, dtype=float)
```
But because I'm trying to learn this new pymeasure instrument architecture, I want to figure out how to combine this with the Instrument.measure() call.  

In my second example, the data comes back like this '-1.000009298 0.000014935,  -1.000000119 0.000027610, ..., -1.000010013 0.000025652' where each pair of numbers represents real and imaginary pair.  There is a variation on this where it's a list of strings with each string containing a pair of numbers (as strings).  Again, I naively would process the data so that it is a single array of complex numbers.  

What is the recommended practice here?

**Comment by jwbrooks0** [comment]
This question is related to the recent pull request: #899

**Comment by BenediktBurger** [comment, marked as answer]
I just wrote an answer with an example and then it vanished, while getting this link for more information:

https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/properties.html

So, the short form is: use `get_process` parameter to specify a callable (method) to process the data you received.

**Comment by jwbrooks0** [comment]
That was it.  Thanks!
